crates/rpc/rpc-types-compat/src/engine/payload.rs

```diff
@@ -83,6 +84,15 @@ pub fn try_payload_v3_to_block(payload: ExecutionPayloadV3) -> Result<Block, Pay
     Ok(base_block)
 }
 
+/// Converts [ExecutionPayloadV4] to [Block]
+pub fn try_payload_v4_to_block(payload: ExecutionPayloadV4) -> Result<Block, PayloadError> {
+    // this performs the same conversion as the underlying V3 payload.
+    //
+    // the new request lists (`deposit_requests`, `withdrawal_requests`) are EL -> CL only, so we do
+    // not do anything special here to handle them
+    try_payload_v3_to_block(payload.payload_inner)
+}
+
 /// Converts [SealedBlock] to [ExecutionPayload]
 pub fn block_to_payload(value: SealedBlock) -> ExecutionPayload {
```

Comment:
this function needs adjustment as well

Reply:
fn seems unused?

Reply:
used in reorg sim, will adjust it in #8053